style/javascript.md

```diff
@@ -1,340 +1,65 @@
+Don't worry about the import order, we prefer using a tool to handle this for us, like:
 
-```js
-const i18n = require("../shared-package/i18n.js");
-```
+* [Prettier Sort Imports Plugin](https://github.com/IanVS/prettier-plugin-sort-imports)
+* [ESLint Sort Imports](https://eslint.org/docs/latest/rules/sort-imports)
 
-The reasoning for this is that our build process that updates translations
-looks for very specific markup, and ES2015 modules get transpiled by Babel
-to an incompatible syntax.
-
-**Grouping**
-There should be 3 clusters of imports: third-party (aka vendor) libraries,
-first-party libraries, and Flow types.
-
-This approximately mirrors our [Python import style](python.md#import-style),
-though there are no "system" imports in JavaScript.
-
-"First party" code is anything we wrote whose primary source lives in
-the repository its being used in.  Underscore is third party because
-we didn't write it.  KaTeX is third party in webapp because even
-though we wrote it, its primary sources lives in a different
-repository.
-
-**Named Imports**
-Named imports should go on the same line, when possible. When 3+ named
-imports are imported, break each named import onto its own line.
-
-Yes:
-```js
-import $ from "jquery";
-import Kicksend from "../../third_party/javascript-khansrc/mailcheck/mailcheck.js";
-import React, {Component} from "react";
-import _ from "underscore";
-
-import APIActionResults from "../shared-package/api-action-results.js";
-import Cookies from "../shared-package/cookies.js";
-import DashboardActions from './datastores/dashboard-actions.js';
-import HappySurvey from "../missions-package/happy-survey.jsx";
-import UserMission from "../missions-package/user-mission.js";
-import cookieStoreRenderer from "../shared-package/cookie-store.handlebars";
-import mainThing, {
-    firstOtherThing,
-    secondOtherThing,
-    thirdOtherThing,
-} from '../somewhere';
-
-import type {Thing} from '../somewhere-else';
-```
-
-No:
-```js
-// Avoid require()
-var _ = require("underscore");
-
-// Avoid grouping first- and third-party libraries
-import React from 'react';
-import APIActionResults from "../shared-package/api-action-results.js";
-
-// Avoid breaking named imports onto multiple lines when possible.
-import React, {
-    Component
-} from 'react';
-
-```
+----------
 
-------------------------------
-### Comments and documentation
+## Comments and documentation
 
-#### Inline Comments
+### Inline Comments
 
 Inline style comments should be of the `//` variety, not the `/* */`
 variety.
 
-#### Top level file and class comments
+### Top level file, class, and component comments
 
-All files and classes should have JSDoc comments.
-
-JSDoc can be parsed by a number of open source tools, and must be well-formed.
+All files, classes, and components should have a comment.
```

Comment:
**suggestion:** Does it make sense to say that all files should have a comment? If they only export a class/component and that has a comment, why have the effort of adding another comment to maintain?

Reply:
Good call, added a bit clarifying that we only need one comment per file/class/component.